### ISSUE TITLE
Cannot start gta, it may be because the rockstar launcher cannot start

### ISSUE THREAD

**Issue by YuHuanTin** [sandboxie-plus/Sandboxie#2720]
### Describe what you noticed and did

I'm trying to run the gta v downloaded from steam in the sandbox (standard isolation type), but I can't start the game

- This is the message box that I displayed when launching the game directly from steam (Cannot launch the game, please verify your game data)
![when i start from steam](https://user-images.githubusercontent.com/51024916/221408560-6af08f0a-9c0e-4d61-a06a-deff88817f37.png)
- This is the message box that pops up when I try to launch LauncherPatcher.exe (rockstart launcher) (Failed to initialize Rockstar Games Launcher)
![image](https://user-images.githubusercontent.com/51024916/221408566-3085dfbe-a5bc-4178-95c8-36e59c758e41.png)

### How often did you encounter it so far?

every time

### Affected program

LauncherPatcher.exe ? this is just my guess

### Download link

Not available

### Where is the program located?

The program is installed only inside a sandbox (NOT in the real system anyway).

### Expected behavior

Successfully start the rockstar launcher and start the game

### What is your Windows edition and version?

Windows 10 Pro 1909 64bit

### In which Windows account you have this problem?

Not relevant to my request.

### Please mention any installed security software

Huorong

### What version of Sandboxie are you running?

Sandboxie 1.7.2 64 bit

### Is it a new installation of Sandboxie?

I just updated Sandboxie from a previous version (to be specified).

### In which sandbox type you have this problem?

In a standard isolation sandbox (yellow sandbox icon).

### Can you reproduce this problem on a new empty sandbox?

Not relevant to my request.

### Sandboxie.ini configuration

```shell
#
# Sandboxie-Plus configuration file
#

[GlobalSettings]
FileRootPath=F:\Sandbox\Root\%SANDBOX%
KeyRootPath=\REGISTRY\USER\Sandbox_%USER%_%SANDBOX%
IpcRootPath=\Sandbox\%USER%\%SANDBOX%\Session_%SESSION%
Template=WindowsRasMan
Template=WindowsLive
Template=OfficeLicensing
Template=InternetDownloadManager
DefaultBox=DefaultBox

[UserSettings_278203E6]
SbieCtrl_AutoStartAgent=SandMan.exe
SbieCtrl_EnableAutoStart=y

[DefaultBox]
Enabled=y
BlockNetworkFiles=y
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
BorderColor=#00ffff,ttl,6
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=AutoRecoverIgnore
ConfigLevel=9
BoxNameTitle=n
MsiInstallerExemptions=y
CopyLimitKb=81920
ClosedFilePath=<BlockNetDevices>,InternetAccessDevices
AllowRawDiskRead=y
FakeAdminRights=y

[Game]
Enabled=y
BlockNetworkFiles=y
RecoverFolder=%{374DE290-123F-4565-9164-39C4925E467B}%
RecoverFolder=%Personal%
RecoverFolder=%Desktop%
BorderColor=#02f6f6,ttl,6
Template=OpenBluetooth
Template=SkipHook
Template=FileCopy
Template=qWave
Template=BlockPorts
Template=LingerPrograms
Template=AutoRecoverIgnore
ConfigLevel=9
AutoRecover=y
BoxNameTitle=n
CopyLimitKb=81920
ClosedFilePath=<BlockNetDevices>,InternetAccessDevices
OpenDevCMApi=y
RpcMgmtSetComTimeout=n
AllowNetworkAccess=<BlockNetAccess>,n
```

**Comment by leonard-lennart** [comment]
I can confirm I have the same issues, starting from 28 Feb 2023, which is when Rockstar released an update for GTA V.

TL;DR: I have not yet found a solution.

Previously, I could run the game normally inside a sandbox. It would be necessary to add a line in the ini section:
StartService=Rockstar Service
I added this line to each of the sandboxes in which I wish to run GTA V. This basically ensures that this dependency program/file is running, for Rockstar Games Launcher (RGL).

However, after the update on the 28th, RGL has been unable to run correctly.
I tried starting from an empty sandbox with the newest version of RGL (1.0.0.14) installed on my computer. I get the same error as one shown by this thread's author.

I tried starting from an empty sandbox with an older version of RGL. The launcher would start, and then automatically update. After the update, the launcher would run normally, until I attempt to start GTA V, and then it would tell me the launcher is out of date, and then fail to install a newer one.

Back the the previous bit. When I start the newest version of RGL inside a sandbox, I get this log file:
[2023-03-02 20:45:51.643] [DISPLAY] LauncherPatcher version: 1.0.0.14
[2023-03-02 20:45:51.658] [DISPLAY] Launcher version: 1.0.0.14
[2023-03-02 20:45:51.658] [DISPLAY] Retrieving Launcher manifest: "https://gamedownloads.rockstargames.com/public/prelauncher.ini"
[2023-03-02 20:45:51.705] [DISPLAY] Retrieving minimum supported versions.
[2023-03-02 20:45:52.362] [DISPLAY] Minimum LauncherPatcher version: 1.0.0.12
[2023-03-02 20:45:52.409] [DISPLAY] Running Launcher from "C:\Program Files\Rockstar Games\Launcher\Launcher.exe"
[2023-03-02 20:46:18.181] [DISPLAY] LauncherPatcher version: 1.0.0.14
[2023-03-02 20:46:18.196] [DISPLAY] Launcher version: 1.0.0.14
[2023-03-02 20:46:18.196] [DISPLAY] Launcher exited with code: -1073741819
[2023-03-02 20:46:18.196] [ ERROR ] Failed to run Launcher
[2023-03-02 20:46:18.243] [ ERROR ] Displaying Error titled: Error
[2023-03-02 20:46:18.243] [ ERROR ]  The Rockstar Games Launcher exited unexpectedly.
Please visit https://support.rockstargames.com for more information.
[2023-03-02 20:46:27.269] [DISPLAY] Shutting down.

I noticed the error code -1073741819, which is the same every time I check. I searched and learnt that this corresponds to the hexadecimal error code 0xC0000005. Most sources I read suggest this is an access violation. Some may suggest this may be something else.

**Comment by leonard-lennart** [comment]
After playing around a little bit. I have identified the offending program to be **Launcher.exe**.

Here are some of my reasons:
- When I try to start RGL (up-to-date version) in Sandboxie, there is always WerFault.exe attached to Launcher.exe.
- I tried, for the up-to-date version of RGL, replacing Launcher.exe with an older version. When starting, WerFault.exe no longer comes up. However, RGL would not start.

**Comment by YuHuanTin** [comment]
> After playing around a little bit. I have identified the offending program to be **Launcher.exe**.
> 
> Here are some of my reasons:
> 
> * When I try to start RGL (up-to-date version) in Sandboxie, there is always WerFault.exe attached to Launcher.exe.
> * I tried, for the up-to-date version of RGL, replacing Launcher.exe with an older version. When starting, WerFault.exe no longer comes up. However, RGL would not start.
> 
> Update: I tried running this program with Windows Debugger. It also gives me the error "Access violation - code c0000005 (first chance)".

I'm sure I have the same problem as you and can't solve it until now

**Comment by isaak654** [comment]
It might be worth a try to enable the Debug tab in Sandboxie Plus if you want to experiment a bit more: https://github.com/sandboxie-plus/Sandboxie/issues/2134#issuecomment-1215466315

**Comment by YuHuanTin** [comment]
> > I added it to the ini but the tab doesn't seem to appear?
> 
> ![Debug_tab](https://user-images.githubusercontent.com/12372772/231220320-8eb749a0-457f-441a-81b5-5592b7b9c6d0.png)

Yes, I can now find the debug interface, but how do I determine what is affecting the program?

**Comment by isaak654** [comment]
> Yes, I can now find the debug interface, but how do I determine what is affecting the program?

As far as I know, there isn't a proven method, it's mainly a matter of research and trial and error: [CONTRIBUTING.md#how-to-help-with-sandboxie-issues](https://github.com/sandboxie-plus/Sandboxie/blob/master/CONTRIBUTING.md#how-to-help-with-sandboxie-issues)

**Comment by YuHuanTin** [comment]
> > Yes, I can now find the debug interface, but how do I determine what is affecting the program?
> 
> As far as I know, there isn't a proven method, it's mainly a matter of research and trial and error: [CONTRIBUTING.md#how-to-help-with-sandboxie-issues](https://github.com/sandboxie-plus/Sandboxie/blob/master/CONTRIBUTING.md#how-to-help-with-sandboxie-issues)

I've tried all the radio options, but nothing seems to work

**Comment by isaak654** [comment]
> I've tried all the radio options, but nothing seems to work

Just in case, more debugging and compatibility improvements are planned in the [current roadmap](https://web.archive.org/web/20230425190420/https://www.wilderssecurity.com/threads/sandboxie-roadmap.445545/).

**Comment by YuHuanTin** [comment]
> > I've tried all the radio options, but nothing seems to work
> 
> Just in case, more debugging and compatibility improvements are planned in the [current roadmap](https://web.archive.org/web/20230425190420/https://www.wilderssecurity.com/threads/sandboxie-roadmap.445545/).

Maybe I should wait until compatibility is enhanced enough to fix this issue 😢

**Comment by Canplayer** [comment]
The same problem also occurred in Red Dead Redemption 2, and the game could not be launched.

**Comment by skflowne** [comment]
I'm also encountering the same error but using Windows Sandbox
